upstream-sha256: 3a3c4064367cb08472a9622a2b6dcb03511b6ee11b9eec86159c73e2ff915240
actual-sha256: 8641e6c760f7b056ac4791fa26b1ffaff3b7c1d9c8e6488031dc408086f258b1
--- upstream.expected
+++ tolk.actual
@@ -1,14 +1,14 @@
 #include <metal_stdlib>
 using namespace metal;
-struct args_t { device int* data0_12; device int* data1_2; device unsigned char* data2_24; };
+struct args_t { device int* data0_12; device int* data1_2; device uchar* data2_24; };
 kernel void E_2_3_2(constant args_t& args [[buffer(0)]], uint3 gid [[threadgroup_position_in_grid]], uint3 lid [[thread_position_in_threadgroup]]) {
-  device int* data0_12 = args.data0_12; device int* data1_2 = args.data1_2; device unsigned char* data2_24 = args.data2_24;
+  device int* data0_12 = args.data0_12; device int* data1_2 = args.data1_2; device uchar* data2_24 = args.data2_24;
   int2 val0 = (*((device int2*)((data1_2+0))));
   int gidx0 = gid.x; /* 2 */
   int lidx0 = lid.x; /* 3 */
   int alu0 = (lidx0+(gidx0*3));
-  unsigned char val1 = (((-1<val0.x)&(val0.x<4))?*(data2_24+(alu0+(val0.x*6))):((unsigned char)(0u)));
-  unsigned char val2 = (((-1<val0.y)&(val0.y<4))?*(data2_24+(alu0+(val0.y*6))):((unsigned char)(0u)));
+  uchar val1 = (((-1<val0.x)&(val0.x<4))?*(data2_24+(alu0+(val0.x*6))):((uchar)(0u)));
+  uchar val2 = (((-1<val0.y)&(val0.y<4))?*(data2_24+(alu0+(val0.y*6))):((uchar)(0u)));
   *(data0_12+alu0) = ((int)(val1));
   *(data0_12+(alu0+6)) = ((int)(val2));
 }
@@ -30,33 +30,33 @@
 ---
 #include <metal_stdlib>
 using namespace metal;
-struct args_t { device float* data0_384; device int* data1_2; device unsigned char* data2_384; device float* data3_12; };
+struct args_t { device float* data0_384; device int* data1_2; device uchar* data2_384; device float* data3_12; };
 kernel void E_3_32_2_2(constant args_t& args [[buffer(0)]], uint3 gid [[threadgroup_position_in_grid]], uint3 lid [[thread_position_in_threadgroup]]) {
-  device float* data0_384 = args.data0_384; device int* data1_2 = args.data1_2; device unsigned char* data2_384 = args.data2_384; device float* data3_12 = args.data3_12;
+  device float* data0_384 = args.data0_384; device int* data1_2 = args.data1_2; device uchar* data2_384 = args.data2_384; device float* data3_12 = args.data3_12;
   int2 val0 = (*((device int2*)((data1_2+0))));
   int gidx0 = gid.x; /* 3 */
   int lidx0 = lid.x; /* 32 */
   int alu0 = (lidx0+(gidx0<<5));
-  unsigned char val1 = (((-1<val0.x)&(val0.x<4))?*(data2_384+(alu0+(val0.x*96))):((unsigned char)(0u)));
-  unsigned char val2 = (((-1<val0.y)&(val0.y<4))?*(data2_384+(alu0+(val0.y*96))):((unsigned char)(0u)));
+  uchar val1 = (((-1<val0.x)&(val0.x<4))?*(data2_384+(alu0+(val0.x*96))):((uchar)(0u)));
+  uchar val2 = (((-1<val0.y)&(val0.y<4))?*(data2_384+(alu0+(val0.y*96))):((uchar)(0u)));
   int alu1 = ((gidx0<<1)+(lidx0>>4));
   float val3 = (*(data3_12+(alu1+6)));
   float val4 = (*(data3_12+alu1));
-  unsigned char alu2 = (val1>>((unsigned char)(4u)));
-  float cast0 = ((float)((val1&((unsigned char)(7u)))));
-  float cast1 = ((float)((alu2&((unsigned char)(7u)))));
+  uchar alu2 = (val1>>((uchar)(4u)));
+  float cast0 = ((float)((val1&((uchar)(7u)))));
+  float cast1 = ((float)((alu2&((uchar)(7u)))));
   float alu3 = ((cast0<7.0f)?(cast0+-2.0f):6.0f);
   float alu4 = ((cast0<5.0f)?(cast0*0.5f):alu3);
   float alu5 = ((cast1<7.0f)?(cast1+-2.0f):6.0f);
   float alu6 = ((cast1<5.0f)?(cast1*0.5f):alu5);
-  unsigned char alu7 = (val2>>((unsigned char)(4u)));
-  float cast2 = ((float)((val2&((unsigned char)(7u)))));
-  float cast3 = ((float)((alu7&((unsigned char)(7u)))));
+  uchar alu7 = (val2>>((uchar)(4u)));
+  float cast2 = ((float)((val2&((uchar)(7u)))));
+  float cast3 = ((float)((alu7&((uchar)(7u)))));
   float alu8 = ((cast2<7.0f)?(cast2+-2.0f):6.0f);
   float alu9 = ((cast2<5.0f)?(cast2*0.5f):alu8);
   float alu10 = ((cast3<7.0f)?(cast3+-2.0f):6.0f);
   float alu11 = ((cast3<5.0f)?(cast3*0.5f):alu10);
   int alu12 = ((gidx0<<6)+(lidx0<<1));
-  *((device float2*)((data0_384+(alu12+192)))) = float2((alu9*(1.0f+(((float)(((val2&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val3),(alu11*(1.0f+(((float)((alu7>>((unsigned char)(3u)))))*-2.0f))*val3));
-  *((device float2*)((data0_384+alu12))) = float2((alu4*(1.0f+(((float)(((val1&((unsigned char)(15u)))>>((unsigned char)(3u)))))*-2.0f))*val4),(alu6*(1.0f+(((float)((alu2>>((unsigned char)(3u)))))*-2.0f))*val4));
+  *((device float2*)((data0_384+(alu12+192)))) = float2((alu9*(1.0f-(((float)(((val2&((uchar)(15u)))>>((uchar)(3u)))))*2.0f))*val3),(alu11*(1.0f-(((float)((alu7>>((uchar)(3u)))))*2.0f))*val3));
+  *((device float2*)((data0_384+alu12))) = float2((alu4*(1.0f-(((float)(((val1&((uchar)(15u)))>>((uchar)(3u)))))*2.0f))*val4),(alu6*(1.0f-(((float)((alu2>>((uchar)(3u)))))*2.0f))*val4));
 }
